### PR TITLE
57 flexible configuration

### PR DESCRIPTION
**Pull request by rock-yu**
## Proposed changes

Implemented "scrutineer2" to run verification with a pair of stream configurations (existing bin/scrutineer is still supported)
```
scrutineer/target/appassembler/bin/scrutineer2 \
    --primary-config=jdbc.properties \
    --secondary-config=elasticsearch7.properties \
    --numeric
```

**This addresses issue #57 **

## Checklist

- [x] I've read the [CONTRIBUTING](https://github.com/Aconex/scrutineer/blob/master/CONTRIBUTING.md) doc
- [x] All tests pass locally with my changes
- [x] I've added unit tests and integration tests.
- [x] I've added necessary documentation (if appropriate)

## Further comments
Let's start the review/discussion first, I have updated the README, CHANGELOG

### CODE REVIEW

File: scrutineer/src/main/java/com/aconex/scrutineer/v2/ScrutineerCommandLineOptionsV2Extension.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    private Map<String, String> readPropertyFromClassPath(String configProperty) {
+        try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(configProperty)) {
+            if (is == null) {
+                throw new IllegalArgumentException(format("config property file: '%s' not found in classpath, stream config files should be placed under 'config' folder", configProperty));
+            }
+
+            Properties properties = new Properties();
+            properties.load(is);
+            return new HashMap(properties);
```
Comment (tallpsmith) [inline scrutineer/src/main/java/com/aconex/scrutineer/v2/ScrutineerCommandLineOptionsV2Extension.java:54]: Should we consider making this an immutable Map?

**Review by tallpsmith — COMMENTED** [review]
Can you pull out the `TODOs` into issues and remove from the code?

**Comment by tallpsmith** [comment]
Also if you can rebase as I've added security checks to the Github workflows set that you can bring in.

File: scrutineer/config/elasticsearch7.properties [hunk 2]
```diff
@@ -0,0 +1,9 @@
+stream.connector.class=com.aconex.scrutineer.elasticsearch.v7.ElasticSearchStreamConnector
```
[the remark this answers is not in the document]
Comment (rock-yu) [reply scrutineer/config/elasticsearch7.properties:1]: Yes they are. Same way kafka-connector ship their configurations. https://github.com/confluentinc/kafka-connect-jdbc/tree/master/config

[the remark this answers is not in the document]
Comment (rock-yu) [reply scrutineer/config/elasticsearch7.properties:1]: I think we should add separate documentation for each field we support

File: scrutineer/src/main/java/com/aconex/scrutineer/v2/ScrutineerCommandLineOptionsV2Extension.java (hunk 1 above)
Comment (rock-yu) [reply scrutineer/src/main/java/com/aconex/scrutineer/v2/ScrutineerCommandLineOptionsV2Extension.java:54]: Good point, will do

File: scrutineer/config/elasticsearch7.properties (hunk 2 above)
[the remark this answers is not in the document]
Comment (rock-yu) [reply scrutineer/config/elasticsearch7.properties]: I've rename them to _example-elasticsearch7.properties_ and _example-jdbc.properties_ so user won't need to second guess them

[the remark this answers is not in the document]
Comment (tallpsmith) [reply scrutineer/config/elasticsearch7.properties]: yeah that's a much better idea, thanks.

**Comment by rock-yu** [comment]
I have updated README AND CHANGELOG with instructions to run scrutineer2

File: README.md [hunk 3]
```diff
@@ -90,6 +92,60 @@ package.  We already have a JTDS driver in there if you're using SQL Server (tha
...
+                --secondary-config=elasticsearch7.properties \
+                --numeric
+
+* **primary-config** - the configuration file for the primary stream
+* **secondary-config** - the configuration file for the secondary stream
+* **numeric** - use this if your query returns results numerically ordered
+
+
+
+Provide `conf/jdbc.properties` to create the primary stream
```
Comment (tallpsmith) [inline README.md:111]: Probably worth mentioning here that the primary & secondary streams can be ANY type of configuration, not just the example here (though this is the common pattern).

File: README.md [hunk 4]
```diff
@@ -90,6 +92,60 @@ package.  We already have a JTDS driver in there if you're using SQL Server (tha
...
+Also provide `conf/elasticsearch7.properties` to create the secondary stream
+```
+stream.connector.class=com.aconex.scrutineer.elasticsearch.v7.ElasticSearchStreamConnector
+es.cluster.name=myCluster
+es.hosts=localhost:9300
+es.index.name=TestIndex
+es.query=*
+```
+
+* **stream.connector.class** – The connector class to use to create the stream
```
Comment (tallpsmith) [inline README.md:140]: Can you create a local URL reference to the 2 example configuration files that outline ALL the config options in the README?  For example, doesn't the ES one have Authentication support already?  Just want to link the documentation.

File: scrutineer/src/main/java/com/aconex/scrutineer/v2/ScrutineerCommandLineOptionsV2Extension.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        return readPropertyFromClassPath(commandLineOptions.primaryConfig);
+    }
+
+    @Override
+    public Map<String, String> getSecondaryConnectorConfigs() {
+        return readPropertyFromClassPath(commandLineOptions.secondaryConfig);
+    }
+
+    private Map<String, String> readPropertyFromClassPath(String configProperty) {
+        try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(configProperty)) {
```
Comment (tallpsmith) [inline scrutineer/src/main/java/com/aconex/scrutineer/v2/ScrutineerCommandLineOptionsV2Extension.java:48]: ok so this is _requiring_ that the config file be part of the class path?  We should document that requirement in the `README` (it wasn't clear what the relative reference is in the docs there)

**Review by tallpsmith — CHANGES_REQUESTED** [review]
Looking great, just a tiny set of changes and we're done.

File: README.md (hunk 3 above)
Comment (rock-yu) [reply README.md]: Good point

File: README.md (hunk 4 above)
Comment (rock-yu) [reply README.md]: DONE, user can click the link to see the full list of configuration and definitions for each field (the README is a lot less noisy now)

File: scrutineer/src/main/java/com/aconex/scrutineer/v2/ScrutineerCommandLineOptionsV2Extension.java (hunk 5 above)
Comment (rock-yu) [reply scrutineer/src/main/java/com/aconex/scrutineer/v2/ScrutineerCommandLineOptionsV2Extension.java:48]: Yes, `config` was added to the classpath.

I've updated the README: 

* **primary-config** - the configuration file for the primary stream, placed under `config` folder

**Review by tallpsmith — APPROVED** [review]

**Pull request merged**